src/background/utils/network.ts

```diff
@@ -9,22 +9,56 @@ interface RequestParams {
 
 export async function readText(params: RequestParams): Promise<string> {
     return new Promise((resolve, reject) => {
-        const request = new XMLHttpRequest();
-        request.overrideMimeType('text/plain');
-        request.open('GET', params.url, true);
-        request.onload = () => {
-            if (request.status >= 200 && request.status < 300) {
-                resolve(request.responseText);
-            } else {
-                reject(new Error(`${request.status}: ${request.statusText}`));
+        if (XMLHttpRequest) {
```

Comment:
Can we move these 2 detections to `utils/platform.ts`

In the form of `const is IsXMLHttpRequestSupported = typeof XMLHttpRequest === 'function'` and `typeof isFetchSupported = typeof fetch === 'function'`.

Also should it by default prefer fetch over XMLHttpRequest?

Reply:
Oops, too late 🤷🏽‍♂️

Reply:
Oh, I rushed and missed your comments by one minute. I'll create a follow-up for this.

Reply:
Here it is: #6486